system/emergency_goal_manager/src/emergency_goal_manager_core.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright 2022 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <emergency_goal_manager_core.hpp>
+
+namespace emergency_goal_manager
+{
+EmergencyGoalManager::EmergencyGoalManager() : Node("emergency_goal_manager")
+{
+  // Subscriber
+  sub_emergency_goals_ = create_subscription<tier4_system_msgs::msg::EmergencyGoalsStamped>(
+    "~/input/emergency_goals", rclcpp::QoS{1},
+    std::bind(&EmergencyGoalManager::onEmergencyGoals, this, std::placeholders::_1));
+  sub_emergency_goals_clear_command_ =
+    create_subscription<tier4_system_msgs::msg::EmergencyGoalsClearCommand>(
+      "~/input/emergency_goals_clear_command", rclcpp::QoS{1},
+      std::bind(&EmergencyGoalManager::onEmergencyGoalsClearCommand, this, std::placeholders::_1));
+
+  // Client
+  client_set_mrm_route_points_callback_group_ =
+    create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
+  client_set_mrm_route_points_ = create_client<SetRoutePoints>(
+    "/planning/mission_planning/mission_planner/srv/set_mrm_route",
+    rmw_qos_profile_services_default, client_set_mrm_route_points_callback_group_);
+  client_clear_mrm_route_callback_group_ =
+    create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
+  client_clear_mrm_route_ = create_client<ClearRoute>(
+    "/planning/mission_planning/mission_planner/srv/clear_mrm_route",
+    rmw_qos_profile_services_default, client_clear_mrm_route_callback_group_);
+
+  // Initialize
+  while (!client_set_mrm_route_points_->wait_for_service(std::chrono::seconds(1)) && rclcpp::ok()) {
+  }
+  while (!client_clear_mrm_route_->wait_for_service(std::chrono::seconds(1)) && rclcpp::ok()) {
+  }
+}
+
+void EmergencyGoalManager::onEmergencyGoals(
+  const tier4_system_msgs::msg::EmergencyGoalsStamped::SharedPtr msg)
+{
+  if (!emergency_goals_map_.empty()) {
+    emergency_goals_map_.clear();
+  }
```

Comment:
Is this process as intended?
Every time a goals topic is subscribed, goals is reset for all modules.

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/5634/commits/abfe52ebf02bc6d5ff160f4e3099b0139602a0f0